src/Extensions/Abstractions/Tag.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Omex.Extensions.Abstractions
+{
+	/// <summary>
+	/// Tagging utitilty class
+	/// </summary>
+	public static class Tag
+	{
+		/// <summary>
+		/// Create an EventId using a tag using GitTagger
+		/// </summary>
+		public static EventId ReserveTag(int tagId) => new EventId(tagId);
+
+
+		/// <summary>
+		/// Create an EventId tag using CallerFilePath and CallerLineNumber
+		/// </summary>
+		public static EventId Create(
+			[CallerFilePath] string sourceFilePath = "",
+			[CallerLineNumber] int sourceLineNumber = 0) =>
```

Comment:
Do we really need to default these? Would it not be better just to fail the build if these can't be set?

Or do we anticipate this would lead to erroneous failures? If so, is there anything we can do to mitigate these?

Reply:
I'm afraid having default value is required by this attribute, removing it will lead to the following error:
```
CS4021	The CallerFilePathAttribute may only be applied to parameters with default values
```
I'm not sure why it was decided to do it like this, some details could be found here https://stackoverflow.com/a/58954747/1942296
although I don't find them very convincing, it could be to simplify backward compatibility or due to limitation of compiler implementation at that version.